gtsfm/scene_optimizer.py

```diff
@@ -301,12 +302,15 @@ def visualize_camera_poses(
     fig = plt.figure()
     ax = fig.gca(projection="3d")
 
-    viz_utils.plot_poses_3d(pre_ba_poses, ax, center_marker_color="c", label_name="Pre-BA")
-    viz_utils.plot_poses_3d(post_ba_poses, ax, center_marker_color="k", label_name="Post-BA")
     if gt_pose_graph is not None:
-        gt_pose_graph = comp_utils.align_poses_sim3(post_ba_poses, gt_pose_graph)
+        # ground truth is used as the reference
+        pre_ba_poses = comp_utils.align_poses_sim3(gt_pose_graph, copy.deepcopy(pre_ba_poses))
+        post_ba_poses = comp_utils.align_poses_sim3(gt_pose_graph, copy.deepcopy(post_ba_poses))
```

Comment:
why did you add `deepcopy`. Was it not working as expected earlier?

Reply:
so my concern before was that we were modifying the ground truth, to fit the estimated data. I prefer the other way around.

However, I don't want to modify the original data just for the sake of alignment (since we'll do that later when comparing with ground truth)

So I prefer to just work with a copy of the data

Reply:
Ok got it.